pytorch_lightning/trainer/lr_finder.py

```diff
@@ -0,0 +1,422 @@
+            if hasattr(model.hparams, self.auto_lr_find):
+                setattr(model.hparams, self.auto_lr_find, lr)
+            else:
+                raise MisconfigurationException(
+                    f'`auto_lr_find` was set to {self.auto_lr_find}, however'
+                    ' could not find this as a field in model.hparams.')
+        else:
+            if hasattr(model.hparams, 'lr'):
+                model.hparams.lr = lr
+            elif hasattr(model.hparams, 'learning_rate'):
+                model.hparams.learning_rate = lr
+            else:
+                raise MisconfigurationException(
+                    'When auto_lr_find is set to True, expects that hparams'
+                    ' either has field `lr` or `learning_rate` that can overridden')
+
+    def _model_dump(self, filepath, model):
+        """ Dump model state, for restoring after lr finder """
+        checkpoint = model.state_dict()
+        if self.proc_rank == 0:
+            # do the actual save
+            try:
+                self._atomic_save(checkpoint, filepath)
+            except AttributeError:
+                if 'hparams' in checkpoint:
+                    del checkpoint['hparams']
+
+                self._atomic_save(checkpoint, filepath)
+
+    def _model_restore(self, filepath, model):
+        """ Restore model state """
+        model.load_state_dict(torch.load(str(filepath)))
+
+    def find_lr(self,
+                model: LightningModule,
+                train_dataloader: Optional[DataLoader] = None,
+                min_lr: float = 1e-8,
+                max_lr: float = 1,
+                num_training: int = 100,
+                mode: str = 'exponential',
+                num_accumulation_steps: int = 1):
+        r"""
+        find_lr enables the user to do a range test of good initial learning rates,
+        to reduce the amount of guesswork in picking a good starting learning rate.
+
+        Args:
+            model: Model to do range testing for
+
+            train_dataloader: A PyTorch
+                DataLoader with training samples. If the model has
+                a predefined train_dataloader method this will be skipped.
+
+            min_lr: minimum learning rate to investigate
+
+            max_lr: maximum learning rate to investigate
+
+            num_training: number of learning rates to test
+
+            mode: search strategy, either 'linear' or 'exponential'. If set to
+                'linear' the learning rate will be searched by linearly increasing
+                after each batch. If set to 'exponential', will increase learning
+                rate exponentially.
+
+            num_accumulation_steps: number of batches to calculate loss over.
+
+        Example::
+
+            # Setup model and trainer
+            model = MyModelClass(hparams)
+            trainer = pl.Trainer()
+
+            # Run lr finder
+            LRfinder = trainer.find_lr(model, ...)
```

Comment:
in the docs page you called this `lrfinder` and here it's called `LRfinder`, it'd be nice to have consistency there (and perhaps do `lr_finder` for pythonic naming)